src/server/index.ts

```diff
@@ -672,10 +672,14 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
                 // when dealing with a merge (tail.length > 0) we have to skip hash validation
                 // for the operations that were re-indexed (previous hash becomes invalid due the new position in the history)
                 if (tail.length > 0) {
-                    skipHashValidation = [...invertedTrunk, ...tail].some(
-                        invertedTrunkOp =>
-                            invertedTrunkOp.hash === nextOperation.hash
+                    const sourceOperation = operations.find(
+                        op => op.hash === nextOperation.hash
```

Comment:
this will be checked with the operation id in the future right?
Comparing hashes seems a bit risky

Reply:
at this point, hashes didn't changed. what you get here are the same operations (the ones that you want to apply and the ones that are already applied to the document but in conflict) but reshuffled.
So what we want to know in this part of the flow is if an operation was reshuffled when there's a merge conflict. If the answer is "yes", then we should skip the hash validation (because of the new position of the operation in the operation history).
I think we can get the same result using the new "id" prop, if the original operation (same id) has a different index (after a merge conflict) then we should skip the hash validation.

Reply:
we can change it to use the new "id" prop instead (after implementation), so it's more clear 👍🏻

Reply:
It might happen that two different operations result in the same hash, or an operation doesn't affect the state so it's hash will be the same as the previous operation

Reply:
you're right, then we should change this by the id of the operation